### PR TITLE
Add messages to LdapIdentityStore

### PR DESCRIPTION
**Pull request by ryanesch**
#build

### CODE REVIEW

**Comment by bsbyrd1** [comment]
@rajivsen Hi there! Any questions about my previous comment?

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
#build

**Comment by ryanesch** [comment]
@OpenLiberty/message-reviewer Can I get my new messages reviewed?

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 1]
```diff
@@ -67,3 +67,31 @@ JAVAEESEC_WARNING_NO_PWD.useraction=If a password was expected, review the calle
...
+
+JAVAEESEC_ERROR_EXCEPTION_ON_GROUP_SEARCH=CWWKS3402E: Searching for groups of {0} has failed. The following exception occurred: {1}
+JAVAEESEC_ERROR_EXCEPTION_ON_GROUP_SEARCH.explanation=An exception occurred searching for groups of the specified user.
+JAVAEESEC_ERROR_EXCEPTION_ON_GROUP_SEARCH.useraction=Review the provided exception.
+
+JAVAEESEC_WARNING_MULTI_CALLER_LDAP=CWWKS3403W: Multiple results were returned for the {0} caller on the LdapIdentityStore using the {1} filter and the {2} searchBase.
+JAVAEESEC_WARNING_MULTI_CALLER_LDAP.explanation=Multiple results were returned for the requested caller. The caller search should only return a single result.
+JAVAEESEC_WARNING_MULTI_CALLER_LDAP.useraction=Review the callerSearchFilter, searchBase, and repository contents.
+
+JAVAEESEC_WARNING_EXCEPTION_ON_GETATTRIBUTES=CWWKS3404W: Searching for the {1} attribute on the {0} entity has failed. The following exception occurred: {1}
```
Comment (jvanhill) [inline dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: This is using {1} twice: once for the exception and once for an attribute.

File: dev/com.ibm.ws.security.javaeesec/src/com/ibm/ws/security/javaeesec/identitystore/LdapIdentityStore.java [hunk 2]
```diff
@@ -302,61 +270,41 @@ private String getUserDn(String callerName, String filter, SearchControls contro
...
-        StringBuffer result = new StringBuffer();
-        result.append("[searchScope: ").append(controls.getSearchScope());
-        result.append(", timeLimit: ").append(controls.getTimeLimit());
-        result.append(", countLimit: ").append(controls.getCountLimit());
-        result.append(", returningObjFlag: ").append(controls.getReturningObjFlag());
-        result.append(", returningAttributes: ").append(controls.getReturningAttributes()[0]).append("]");
-        return result.toString();
-    }
-
     private Set<String> getGroups(DirContext context, String userDn) {
```
Comment (jvanhill) [inline dev/com.ibm.ws.security.javaeesec/src/com/ibm/ws/security/javaeesec/identitystore/LdapIdentityStore.java]: Add Javadoc. This applies to the class in general.

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops (hunk 1 above)
Comment (ryanesch) [reply dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Good catch.

File: dev/com.ibm.ws.security.javaeesec/src/com/ibm/ws/security/javaeesec/identitystore/LdapIdentityStore.java (hunk 2 above)
Comment (ryanesch) [reply dev/com.ibm.ws.security.javaeesec/src/com/ibm/ws/security/javaeesec/identitystore/LdapIdentityStore.java]: I added javadoc there and fixed a few places that had outdated parameters.

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 3]
```diff
@@ -67,3 +67,31 @@ JAVAEESEC_WARNING_NO_PWD.useraction=If a password was expected, review the calle
 JAVAEESEC_WARNING_MULTI_CALLER=CWWKS1924W: The ''{1}'' query returned multiple results for the {0} caller on the DatabaseIdentityStore.
 JAVAEESEC_WARNING_MULTI_CALLER.explanation=Multiple results were returned for the requested caller. The caller query should only return a single result.
 JAVAEESEC_WARNING_MULTI_CALLER.useraction=Review the caller query and database contents. Change the caller query to return one result.
+
+JAVAEESEC_ERROR_EXCEPTION_ON_BIND=CWWKS3400W: Binding with the {0} caller has failed. The following exception occurred: {1}
```
Comment (bsbyrd1) [inline dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Change to:
Binding with the {0} caller failed. The following exception occurred: {1}

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 4]
```diff
@@ -67,3 +67,31 @@ JAVAEESEC_WARNING_NO_PWD.useraction=If a password was expected, review the calle
...
+JAVAEESEC_ERROR_EXCEPTION_ON_BIND.explanation=An exception occurred while binding with the provided caller.
+JAVAEESEC_ERROR_EXCEPTION_ON_BIND.useraction=Review the provided exception.
+
+JAVAEESEC_ERROR_EXCEPTION_ON_SEARCH=CWWKS3401E: Searching for the {0} caller with the {1} filter and the {2} searchBase failed. The following exception occurred: {3}
+JAVAEESEC_ERROR_EXCEPTION_ON_SEARCH.explanation=An exception occurred while searching for the provided caller.
+JAVAEESEC_ERROR_EXCEPTION_ON_SEARCH.useraction=Review the provided exception.
+
+JAVAEESEC_ERROR_EXCEPTION_ON_GROUP_SEARCH=CWWKS3402E: Searching for groups of {0} has failed. The following exception occurred: {1}
```
Comment (bsbyrd1) [inline dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Change to:
Searching for groups of {0} failed. The following exception occurred: {1}

File: dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops [hunk 5]
```diff
@@ -67,3 +67,31 @@ JAVAEESEC_WARNING_NO_PWD.useraction=If a password was expected, review the calle
...
+JAVAEESEC_ERROR_EXCEPTION_ON_GROUP_SEARCH.explanation=An exception occurred searching for groups of the specified user.
+JAVAEESEC_ERROR_EXCEPTION_ON_GROUP_SEARCH.useraction=Review the provided exception.
+
+JAVAEESEC_WARNING_MULTI_CALLER_LDAP=CWWKS3403W: Multiple results were returned for the {0} caller on the LdapIdentityStore using the {1} filter and the {2} searchBase.
+JAVAEESEC_WARNING_MULTI_CALLER_LDAP.explanation=Multiple results were returned for the requested caller. The caller search should only return a single result.
+JAVAEESEC_WARNING_MULTI_CALLER_LDAP.useraction=Review the callerSearchFilter, searchBase, and repository contents.
+
+JAVAEESEC_WARNING_EXCEPTION_ON_GETATTRIBUTES=CWWKS3404W: Searching for the {1} attribute on the {0} entity has failed. The following exception occurred: {2}
```
Comment (bsbyrd1) [inline dev/com.ibm.ws.security.javaeesec/resources/com/ibm/ws/security/javaeesec/internal/resources/JavaEESecMessages.nlsprops]: Change to:
Searching for the {1} attribute on the {0} entity failed. The following exception occurred: {2}

**Pull request merged**